packages/x-components/src/components/suggestions/base-suggestion.vue

```diff
@@ -147,30 +171,44 @@
     }
 
     /**
-     * Highlights the matching part of the suggestion query with the query passed as prop of the
-     * component putting it inside a `<span>` tag.
+     * Returns the HTML to render the query, including the matching part and the filter.
      *
      * @remarks
      * The query prop should be normalized.
      *
-     * @returns The suggestion's query with the matching part inside a `<span>` tag.
+     * @returns The suggestion's query HTML.
      * @public
      */
     protected get queryHTML(): string {
-      if (this.hasMatchingQuery) {
-        const matcherIndex = normalizeString(this.suggestion.query).indexOf(this.query);
-
-        const [beginning, matching, end] = this.splitAt(
-          this.suggestion.query,
-          matcherIndex,
-          this.query.length
-        );
-
-        const attrsMatching = 'data-test="matching-part" class="x-suggestion__matching-part"';
-        return `${beginning}<span ${attrsMatching}>${matching}</span>${end}`;
+      let query = this.hasMatchingQuery
```

Comment:
maybe we should just leave it as it was, and create a new span in the slot to add the filter.label